src/events/provider.py

```diff
@@ -75,21 +75,37 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             password = auth.generate_password()
             auth.create_user(username=username, password=password)
 
-            kyuubi_address = service_manager.get_service_endpoint(
+            kyuubi_endpoint = service_manager.get_service_endpoint(
                 expose_external=self.charm.config.expose_external
             )
-            endpoint = f"jdbc:hive2://{kyuubi_address}/" if kyuubi_address else ""
+            jdbc_uri = f"jdbc:hive2://{kyuubi_endpoint}/" if kyuubi_endpoint else ""
 
             # Set the JDBC endpoint.
             self.database_provides.set_endpoints(
                 event.relation.id,
-                endpoint,
+                kyuubi_endpoint,
             )
 
+            # Set the JDBC URI
+            self.database_provides.set_uris(event.relation.id, jdbc_uri)
```

Comment:
Here I have proceeded with setting `hostname:port` combination as the "endpoint" and the `jdbc:hive2://hostname:port/` as the "uri" being consistent with other database charms.

Reply:
I am a little confused by this because I thought that we were all migrating from `uris` to `endpoints`, not keeping the two of them

Reply:
I would have preferred to keep the `uri`. The URI seems better than endpoint since it also has the protocol there. Is there a spec about this? I'm not sure I recall a discussion about this, and I'm not aware of the idea of dropping URI

@delgod what's your take on this?

Reply:
I found [this document](https://docs.google.com/document/d/1UpIZbBz-af2-uNLVTijdbwuW9ASy4IOE5tU9dVEY1Ho/edit?usp=sharing) which mentions the URI (though not the endpoint). I included the endpoint as well as URI here as well because for clients like beeline, its easier to connect with full connection string provided in URI, whereas for clients like `pyhive`, it would be easier extracting hostname:port from the endpoints than a rather convoluted URI.

Reply:
To give more context, my initial confusion comes from the keys we put in the databags themselves, where we changed `uris` to `endpoints` (see [kafka](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/kafka_client/v0#example), [mysql](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/mysql_client/v0#example) or [mongodb](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/mongodb_client/v0#example).
But since this line is just an internal implementation detail, maybe it does not matter

Reply:
`endpoints` is mandatory, and `uris` is optional.
endpoints have strict format `host1,host2` or `host1:port1,host2:port2`.
and `uris` is something which can be passed to the driver, as it is, without additional parsing or modification.

Reply:
Thank you Mykola. The PR in its current state is good to be merged, then. Merging it now :)